### ISSUE TITLE
Madmin: Spawnpoint Statistics returning incomplete results (fence names issue)

### ISSUE THREAD

**Issue by TiMXL73** [Map-A-Droid/MAD#663]
**SYMPTOM:**
In madmin, `Spawnpoint Statistics` is returning only 1 fence labeled `unnamed` for me and partial results for others.

**ISSUE:**
Madmin is still looking for the `[some_geo_name]` line above the coordinates in the `fence_data` field to name its fences.

**EXPECTED BEHAVIOR:**
Now that everything has been moved to the database, madmin should no longer be looking for that line and instead use the mandatory field `name` as configured in madmin.

**HOW TO REPRODUCE:**	
`Madmin>Settings>Geofences`, if you only set coordinates in `fence_data`, that geofence will be returned as `unnamed`.

`Madmin>Settings>Geofences`, if you add `[some_geo_name]` before your coordinates in `fence_data`, that geofence will be returned as `some_geo_name` regardless of its actual name in madmin.

**THOUGHTS / DISCUSSION:**
Considering the `name` field is mandatory when creating a geofence, I do not believe that `[some_geo_name]` should still be allowed/used/saved in the `fence_data` field as it can only lead to confusion if `[some_geo_name]` and `name` of the same fence are different or if multiple fences have the same `[some_geo_name]` for some reason could cause an issue somewhere like this stats page.

Proposal: If `[some_geo_name]` already exists in the `fence_data` field (previous import), simply ignore that information and remove it permanently when saving a geofence to the database unless sees a valid reason to keep it (I don’t).

Once the best course of action has been determined by our overlords, someone well versed in Python will be able to implement a permanent solution.

**TEMP FIX IF YOU NEED A WORKING SPAWNPOINTS STATS PAGE NOW:**
In `geofenceHelper.py`, just change `'name': 'unnamed', ` to `'name': geo_resource['name'],` and it’ll use the DB name if none is present in `fence_data`.

**Comment by ChaosEmpire** [comment]
i run into this issue before, and form my point of view, the area name should get added into the db like beofre
as you can add severeal fences into one

if you name it like one, you cant have anylonger differennt names for quest drop down while having several areas in one geofence,

**Comment by TiMXL73** [comment]
Thanks for the input ChaosEmpire, I completely forgot you could have multiple complete geofences inside a single defined geofence.  This means the `[some_geo_name]` tag cannot be removed or it’ll break that functionality.

Which leaves the question of how to handle the `unnamed`:
-	Always use the DB geofence `name` field? No `unnamed` or duplicate possible.
-	Change nothing and remind users to have a `[some_geo_name]` tag or face possible issues like `unnamed` or duplicate names?
-	Use the first `[some_geo_name]` like now and fallback to DB geofence `name` field if it doesn’t exist (like the suggested temp fix)? The potential for duplicate name would still exist.
 -	Other?

**Comment by krzee** [comment]
in other:
require [name] and give an error if it is not present

**Comment by ChaosEmpire** [comment]
i would set as default the name of the geofence you used when creating the fence by clicking "save to mad"

**Comment by krzee** [comment]
so add [name] when adding if its not there using the "name" entered in to the gui , ya i like that

**Comment by TiMXL73** [comment]
Yup. After messing around with multi-fence in a single declaration that I hadn't considered initially, I think it's the best course of action and exactly what the temp fix is doing.
